src/handlers/search.handler.ts

```diff
@@ -0,0 +1,23 @@
+import Restaurant from "../models/restaurant.model";
+import Dish from "../models/dish.model";
+import Chef from "../models/chef.model";
+
+const SearchHandler = {
+  async search(keyword: string) {
+    const [restaurants, chefs, dishes] = await Promise.all([
+      Restaurant.find({ title: { $regex: keyword, $options: "i" } }),
+      Chef.find({ title: { $regex: keyword, $options: "i" } }),
+      Dish.find({ title: { $regex: keyword, $options: "i" } }),
+    ]);
+
+    const searchResults = {
+      restaurants: restaurants,
+      chefs: chefs,
+      dishes: dishes,
+    };
```

Comment:
If the key and the value are the same, no need to explicit write them but only the key.

```suggestion
    const searchResults = {
      restaurants,
      chefs,
      dishes,
    };
```